scripts/backend.py

```diff
@@ -169,6 +238,312 @@ def page_not_found(error):
+    for o_id in stop_ids:
+        client.disableoutput(o_id)
+
+    LIVESTREAM_INFO['active'] = True
+    LIVESTREAM_INFO['last_played'] = []
+
+
+def stop_streaming():
+    if not LIVESTREAM_INFO['active']:
+        return
+    try:
+        client = MPDClient()
+        client.connect(args["--mpd-host"], LIVESTREAM_INFO['PORT'])
+    except:
+        print('couldn\'t connect to mpd')
+        return
+    c_outputs = client.outputs()
+    start_ids = [c['outputid'] for c in c_outputs if 'cyberia' in c['outputname']]
+    for o_id in start_ids:
+        client.enableoutput(o_id)
+    client.pause(0)
+
+    LIVESTREAM_INFO['active'] = False
+    LIVESTREAM_INFO['last_played'] = []
+    LIVESTREAM_INFO['current_dj'] = None
+
+
+@app.route("/dj/start_streaming")
+@login_required
+def streaming_page():
+    if LIVESTREAM_INFO['current_dj'] is None:
+        LIVESTREAM_INFO['current_dj'] = current_user.id
+        start_streaming()
+        return_text = "stream started, you can connect to icecast now"
+        # return_text = "you should be able to connect to icecast as soon as the currently playing track ends"
+    else:
+        return_text = 'someone else is already streaming..'
+    return return_text
+
+
+# maybe this should happen automatically if playlist update doesn't work for 30 seconds?
+@app.route("/dj/stop_streaming")
+@login_required
+def streaming_over_page():
+    if (LIVESTREAM_INFO['current_dj'] == current_user.id) or (current_user.id == 'admin'):
+        stop_streaming()
+        return_text = 'stream ended'
+    else:
+        return_text = 'nop'
+    return return_text
+
+
+def get_livestream_info():
+    # get current track info
+    req_url = "http://54.208.6.108:8000/status-json.xsl"  # temp dev
+    # req_url = "/radio/status-json.xsl"
+    try:
+        req_info = json.loads(make_requests.get(req_url).content.decode('utf-8'))
+    except Exception as e:
+        print(e)
+        return
+
+    status_metadata = {
+        'artist': '',
+        'title': '',
+        'album': '🎵 L I V E S T R E A M 🎵'
+    }
+
+    stream_data = {
+        'dj_name': '',
+        'dj_pic': '',
+        'stream_desc': '',
+        'live': LIVESTREAM_INFO['active']
+    }
+
+    # fill in with streamer info
+    if stream_data['live']:
+        dj_info = db.get_dj_info(LIVESTREAM_INFO['current_dj'])
+        if dj_info is not None:
+            for k in dj_info:
+                if k in stream_data:
+                    stream_data[k] = dj_info[k]
+            if 'stream_title' in dj_info and len(dj_info['stream_title']) > 0:
+                status_metadata['album'] = dj_info['stream_title']
+
+    if 'source' not in req_info['icestats']:
+        # nothing streaming yet
+        return {'current': status_metadata, 'elapsed': 0, 'stream_data': stream_data}
+
+    sources = req_info['icestats']['source']
+    for source in sources:
+        if 'cyberia' in source['listenurl']:
```

Comment:
I'll add a parameter for the exact mount-name and it'll be an explicit equality check to avoid mishaps with the cyberia-channel.ogg fallback